src/axolotl/utils/ctx_managers/sequence_parallel.py

```diff
@@ -206,12 +207,25 @@ def __init__(
     def __enter__(self):
         # Forward pre-hook to apply sequence parallelism
         def sequence_parallel_pre_hook(_, args, kwargs):
-            # Apply sequence parallelism to kwargs and get original sequence length and padding info
-            kwargs, self.original_seq_len, self.pad_len = (
-                self.apply_sequence_parallelism(batch=kwargs)
+            # Get parameter names from the model's forward function
+            forward_params = list(
+                inspect.signature(self.models[0].forward).parameters.keys()
             )
 
-            return args, kwargs
+            updated_kwargs = kwargs.copy()
+            for i, arg in enumerate(args):
+                if i < len(forward_params):
```

Comment:
What does this check do? Is it for positional args to pass through?

Reply:
Some trainers pass e.g. `input_ids` as an arg rather than a kwarg. This logic handles that case.